Disabled state patterns Doc (#2083)
hashicorp/design-system: website/docs/patterns/disabled-patterns/partials/research.md

## Related articles

> Further reading to help articulate why disabled elements are the least preferable interaction option.

Reviewer: ```suggestion
Further reading on why using disabled elements are the least preferable interaction option.
```
